generator.go

```diff
@@ -90,47 +103,32 @@ func (t *Generator) CreateToken(data Data, options *Option) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	encodedClaim := encode(claimBytes)
 
 	// create the token
-	secureString := fmt.Sprintf("%s%s%s", encodedHeader, TokenSep, encodedClaim)
+	secureString := encodedHeader + TokenSep + encode(claimBytes)
 	signature := sign(secureString, t.secret)
-	token := fmt.Sprintf("%s%s%s", secureString, TokenSep, signature)
+	token := secureString + TokenSep + signature
 
 	if len(token) > 1024 {
-		return "", errors.New("Generated token is too long. The token cannot be longer than 1024 bytes.")
+		return "", ErrTokenTooLong
 	}
 	return token, nil
 }
 
-func encodedHeader() (string, error) {
-	headers := struct {
-		Algorithm string `json:"alg"`
-		Type      string `json:"typ"`
-	}{
-		Algorithm: "HS256",
-		Type:      "JWT",
-	}
-
-	headerBytes, err := json.Marshal(headers)
-	if err != nil {
-		return "", err
-	}
-	return encode(headerBytes), nil
-}
-
 func validate(data Data, isAdmind bool) error {
 	uid, containsID := data["uid"]
+	// TODO: if ther eis no `uid` and is admin, then will fail on the string check...
```

Comment:
if `uid` is not set, then you'd have
```go
uid, containsID := data["uid"] /* "", false */
```

So when you get to L#126 you'd have
```go
_, isString := uid.(string) /* "", false */
if containsID /*false*/ && !isString /* true */ {
...
```

Which would continue through the rest of the function.